modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java

```diff
@@ -423,6 +477,61 @@ public void testWalCompressedBytes() throws Exception {
         assertCorrectWalCompressedBytesMetrics(n1);
     }
 
+    /**
+     * Populates a cache w/32 KB of data
+     *
+     * @param igniteEx - Node
+     */
+    private void populateCache(IgniteEx igniteEx) {
+        for (int i = 0; i < 10; i++)
+            igniteEx.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+    }
+
+    /**
+     * Turns off WAL logging.
+     *
+     * @param igniteEx - Node
+     * @throws IgniteCheckedException
+     */
+    private void turnOffWal(IgniteEx igniteEx) throws IgniteCheckedException {
+        WALDisableContext walDisableCtx = igniteEx.context().cache().context().walState().walDisableContext();
+        assertNotNull(walDisableCtx);
+
+        setFieldValue(walDisableCtx, "disableWal", true);
+
+        assertTrue(walDisableCtx.check());
+        assertNull(walMgr(igniteEx).log(new DataRecord(emptyList())));
+    }
+
+    /**
+     * Check the state of wal archive, and whether the total size of wal (and possibly wal archive) match what is expected.
+     *
+     * @param igniteEx      - Node
+     * @param hasWalArchive - whether wal archiving is enabled
+     * @throws IgniteCheckedException
+     * @throws InterruptedException
+     */
+    private void checkWalArchiveAndTotalSize(IgniteEx igniteEx,
+        boolean hasWalArchive) throws IgniteCheckedException, InterruptedException {
+        FileWriteAheadLogManager walMgr = walMgr(igniteEx);
+
+        SegmentRouter router = walMgr.getSegmentRouter();
+
+        assertEquals(router.hasArchive(), hasWalArchive);
+
+        //wait to avoid race condition where new segments(and corresponding .tmp files) are created after totalSize has been calculated.
+        igniteEx.context().cache().context().database().waitForCheckpoint("test");
```

Comment:
I think it's not good to bypass the race this way.
I still think it is more correct to wait for the initialization of all files in the working directory.